planning/behavior_path_planner/include/behavior_path_planner/scene_module/pull_over/geometric_pull_over.hpp

```diff
@@ -0,0 +1,67 @@
+// Copyright 2022 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef BEHAVIOR_PATH_PLANNER__SCENE_MODULE__PULL_OVER__GEOMETRIC_PULL_OVER_HPP_
+#define BEHAVIOR_PATH_PLANNER__SCENE_MODULE__PULL_OVER__GEOMETRIC_PULL_OVER_HPP_
+
+#include "behavior_path_planner/scene_module/pull_over/pull_over_path.hpp"
+#include "behavior_path_planner/scene_module/pull_over/pull_over_planner_base.hpp"
+#include "behavior_path_planner/scene_module/utils/geometric_parallel_parking.hpp"
+#include "behavior_path_planner/scene_module/utils/occupancy_grid_based_collision_detector.hpp"
+
+#include <autoware_auto_planning_msgs/msg/path_with_lane_id.hpp>
+
+#include <memory>
+#include <vector>
+
+namespace behavior_path_planner
+{
+class GeometricPullOver : public PullOverPlannerBase
+{
+public:
+  explicit GeometricPullOver(
+    rclcpp::Node & node, const PullOverParameters & parameters,
+    const ParallelParkingParameters & parallel_parking_parameters,
+    std::shared_ptr<OccupancyGridBasedCollisionDetector> & occupancy_grid_map, bool is_forward);
```

Comment:
1. add const to `bool is_forward`

2. Any reason to use reference to shared_ptr? Also we can add const here?
Then, it will be `const shared_ptr` without reference.

Reply:
expliciti is not required when the arguments are more than 1.

Reply:
fixed in https://github.com/autowarefoundation/autoware.universe/pull/1911/commits/b47d458c9870cfcb74d01d9866159bd273ee1972